Subject: [PATCH] zlib: enable unaligned memory access

arm32, arm64, x86 and x86_64 are all capable of
handling unaligned memory access.

Enable UNALIGNED_OK in order to improve performance. Define ZLIB_CONST to add constness.

The next_in and msg fields in z_stream will become pointers to const data.

diff --git a/Android.mk b/Android.mk
index 6cbbf18..01bc4d6 100644
--- a/Android.mk
+++ b/Android.mk
@@ -22,9 +22,11 @@ zlib_files := \
 	src/uncompr.c \
 	src/zutil.c
 
+zlib_cflags := -O3 -DUSE_MMAP -DZLIB_CONST -DUNALIGNED_OK
+
 LOCAL_MODULE := libz
 LOCAL_MODULE_TAGS := optional
-LOCAL_CFLAGS += -O3 -DUSE_MMAP
+LOCAL_CFLAGS += $(zlib_cflags)
 LOCAL_CFLAGS_arm64 += -mcpu=generic+crc
 
 # TODO: This is to work around b/24465209. Remove after root cause is fixed
@@ -44,7 +46,7 @@ include $(CLEAR_VARS)
 LOCAL_ARM_MODE := arm
 LOCAL_MODULE := libz
 LOCAL_MODULE_TAGS := optional
-LOCAL_CFLAGS += -O3 -DUSE_MMAP
+LOCAL_CFLAGS += $(zlib_cflags)
 LOCAL_CFLAGS_arm64 += -mcpu=generic+crc
 LOCAL_SRC_FILES := $(zlib_files)
 ifneq ($(TARGET_BUILD_APPS),)
@@ -59,7 +61,7 @@ include $(CLEAR_VARS)
 
 LOCAL_MODULE := libz
 LOCAL_MODULE_TAGS := optional
-LOCAL_CFLAGS += -O3 -DUSE_MMAP
+LOCAL_CFLAGS += $(zlib_cflags)
 LOCAL_SRC_FILES := $(zlib_files)
 LOCAL_MULTILIB := both
 LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)
@@ -71,7 +73,7 @@ include $(CLEAR_VARS)
 
 LOCAL_MODULE := libz-host
 LOCAL_MODULE_TAGS := optional
-LOCAL_CFLAGS += -O3 -DUSE_MMAP
+LOCAL_CFLAGS += $(zlib_cflags)
 LOCAL_SRC_FILES := $(zlib_files)
 LOCAL_MULTILIB := both
 LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)
